nlu.py

```diff
@@ -20,6 +20,38 @@ def _get_commands(text_sequence: [str]) -> set:
     return possible_commands
 
 
+def _consine_similarity(a: [str], b: [str]) -> float:
+    words = set(a+b)
+    vector_a = _word2vector(words, a)
+    vector_b = _word2vector(words, b)
+    return _cosine_theta(vector_a, vector_b)
+
+
+def _cosine_theta(a: [int], b: [int]):
+    numerator = sum(x1 * x2 for x1, x2 in zip(a, b))
+    denominator = _dot(a)*_dot(b)
+    return numerator/denominator
+
+
+def _dot(v: [int]) -> float:
+    return sum(pow(n, 2) for n in v) ** 0.5
```

Comment:
I would say `v in vec` because it's all about sub-vectors in an input vector
```py
import cmath

def _dot(vec: [int]) -> float:
  return cmath.sqrt(
    sum(pow(v, 2) for v in vec
  )
```

And `** 0.5` is not readable, also only working for a positive number which relies on the input generator, you should panic the process if there's negative number or using `sqrt`

Reply:
sure, sqrt is more clear and understanding.

Reply:
BTW, if formatter makes the code be one line, then give sum(...) a temporary variable, else the code would be hard to read

Reply:
I am using `math.sqrt` to instead the `cmath.sqrt`

`cmath.sqrt` return the instance of `complex` and It's not able to operate with `>` `<`